extension/amqp/publisher.go

```diff
@@ -0,0 +1,83 @@
+package amqp
+
+import (
+	"context"
+
+	"github.com/hellofresh/goengine"
+	"github.com/hellofresh/goengine/driver/sql"
+	"github.com/mailru/easyjson"
+	"github.com/streadway/amqp"
+)
+
+var _ sql.ProjectionTrigger = (&NotificationPublisher{}).Publish
+
+// NotificationPublisher is responsible of publishing a notification to queue
+type NotificationPublisher struct {
+	amqpDSN string
+	queue   string
+	logger  goengine.Logger
+
+	connection NotificationConnection
+	channel    NotificationChannel
+}
+
+// NewNotificationPublisher returns an instance of NotificationPublisher
+func NewNotificationPublisher(amqpDSN, queue string,
+	logger goengine.Logger,
+	connection NotificationConnection,
+	channel NotificationChannel,
+) (*NotificationPublisher, error) {
+
+	if _, err := amqp.ParseURI(amqpDSN); err != nil {
+		return nil, goengine.InvalidArgumentError("amqpDSN")
+	}
+	if len(queue) == 0 {
+		return nil, goengine.InvalidArgumentError("queue")
+	}
+	return &NotificationPublisher{
+		amqpDSN:    amqpDSN,
+		queue:      queue,
+		logger:     logger,
+		connection: connection,
+		channel:    channel,
+	}, nil
+}
+
+// Publish sends a ProjectionNotification to Queue
+func (p *NotificationPublisher) Publish(ctx context.Context, notification *sql.ProjectionNotification) error {
+	// Ignore nil notifications since this is not supported
+	if notification == nil {
+		p.logger.Warn("unable to handle nil notification, skipping", nil)
+		return nil
```

Comment:
Should this not return an error?

Reply:
I believe we may want to simply skip `nil` notification instead of throwing an error?

Reply:
I'm not really sure since it could case some serious confusion.... let's at least document this behavior clearly.